toolchain/lexer/tokenized_buffer.cpp

```diff
@@ -716,24 +830,16 @@ auto TokenizedBuffer::Lex(SourceBuffer& source, DiagnosticConsumer& consumer)
   ErrorTrackingDiagnosticConsumer error_tracking_consumer(consumer);
   Lexer lexer(buffer, error_tracking_consumer);
 
+  // Build a table of function pointers that we can use to dispatch to the
+  // correct lexer routine based on the first byte of source text.
```

Comment:
Overall, as a reader of this I'm left wondering why we use a table of function pointers rather than a `switch` on the first character. I think it'd be useful to explain that here.